ReactWindows/ReactNative.Shared/UIManager/NativeViewHierarchyOptimizer.cs

```diff
@@ -477,13 +477,19 @@ private void TransitionLayoutOnlyViewToNativeView(ReactShadowNode node, ReactSty
             // update the layout of this node's children now that it's no 
             // longer layout-only, but we may still receive more layout updates
             // at the end of this batch that we don't want to ignore.
+            // Also <node>.DispatchUpdates optimizes the layout applying out
+            // if screen position/sizes didn't change, yet in this particular case
+            // we want to force the applying of those values since the native view
+            // is a freshly created one with no history.
             ApplyLayoutBase(node);
             for (var i = 0; i < node.ChildCount; ++i)
             {
                 ApplyLayoutBase(node.GetChildAt(i));
             }
 
             _tagsWithLayoutVisited.Clear();
+
+            node.MarkForceLayout();
         }
```

Comment:
Is this a bug in Yoga or a bug in RNW? I don't see this behavior in ReactAndroid, does it also need to be fixed there? [https://github.com/facebook/react-native/blob/4f2cc42a2dfac9d3a799b643c625df7506c4b32c/ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyOptimizer.java](https://github.com/facebook/react-native/blob/4f2cc42a2dfac9d3a799b643c625df7506c4b32c/ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyOptimizer.java)

Reply:
It's not a Yoga bug, Yoga correctly self marks itself as "dirty" during the remove/add child operations above, and later it computes a layout (so node.HasNewLayout becomes true).
The problem is that the applying of the computed values is not gated just by HasNewLayout. That additional optimization checking whether ScreenX & ScreenY changed (presumably from previous layout cycle) is also in place.
Shadow nodes continue to be layout even when they're "layout only", so it's possible ScreenX/ScreenY don't change during a transition, when let's say the reason the view is brought into real world is a change in "opacity", for ex.

Haven't debugged on Android (I've avoided the world of hurt the Android building is so far), I suspect some re-rendering due to layout differences keeps this hidden in the scenarios we tested.
But I don't know for sure.

---
In reply to: [168473459](https://github.com/Microsoft/react-native-windows/pull/1643#discussion_r168473459) [](ancestors = 168473459)